### PR TITLE
Sa/rtl/draft screen rtl fix

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
Fixed icon overlap in draft screen header and switched the back-icon direction in RTL layout.

### Issue number
Draft Screen has some icon overlaps in RTL.

### Screenshots/Video
image

### CODE REVIEW

**Comment by noumantahir** [comment]
@aliseyalvi shouldn't text fields also move to right.....Imagine typing title in Urdu, should not that start from right... ?

**Comment by noumantahir** [comment]
@aliseyalvi also android side seems to be crashing in my tests...

**Comment by aliseyalvi** [comment]
> right aligned editor text field in RTL

Here is the [fix ](https://github.com/ecency/ecency-mobile/pull/2116/commits/a708f9c9298391b8854e3b58ae06fe4cc8debead).
Right aligned the text editor in RTL

**Comment by aliseyalvi** [comment]
> @aliseyalvi shouldn't text fields also move to right.....Imagine typing title in Urdu, should not that start from right... ? Overall I like how neatly we can add support for RTL... good going.

Here is the [fix ](https://github.com/ecency/ecency-mobile/pull/2116/commits/a708f9c9298391b8854e3b58ae06fe4cc8debead).
Right aligned the text editor in RTL

**Comment by aliseyalvi** [comment]
> @aliseyalvi also android side seems to be crashing in my tests...

@noumantahir didn't noticed crash on my side. checked on iOS and android both. Can you please check it again or share some video with me. Working fine on my side btw

**Comment by noumantahir** [comment]
@aliseyalvi I see, apparently my packager was malfunctioning

File: src/utils/I18nUtils.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  if (prevLang != lang) {
+    // if selected lang is RTL, switch the layout
+    if (rtlDetect.isRtlLang(lang)) {
+      await NativeModules.I18nManager.forceRTL(true);
+    } else {
+      await NativeModules.I18nManager.forceRTL(false);
+    }
+
+    Alert.alert(
+      'Restart Ecency?',
```
Comment (noumantahir) [inline src/utils/I18nUtils.ts:15]: @aliseyalvi strings should always be added in `en-US.json` file and referenced using `useInl()`.. for example implementation you can check `commentsView.tsx`

File: src/utils/I18nUtils.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { Alert, NativeModules } from 'react-native';
+import RNRestart from 'react-native-restart';
+import rtlDetect from 'rtl-detect';
+
+export const languageRestart = async (prevLang, lang) => {
+  if (prevLang != lang) {
+    // if selected lang is RTL, switch the layout
+    if (rtlDetect.isRtlLang(lang)) {
```
Comment (noumantahir) [inline src/utils/I18nUtils.ts:8]: may be we should check if layout needs to change based on previous language and only restart if it is required... it seems right now it is asking for restart even layout do not require change

**Review by noumantahir — CHANGES_REQUESTED** [review]

File: src/utils/I18nUtils.ts (hunk 1 above)
Comment (aliseyalvi) [reply src/utils/I18nUtils.ts]: updated. https://github.com/ecency/ecency-mobile/pull/2116/commits/79539637755f2133239ab47f43ff49fcb3b7e6ea

File: src/utils/I18nUtils.ts (hunk 2 above)
Comment (aliseyalvi) [reply src/utils/I18nUtils.ts:8]: updated. https://github.com/ecency/ecency-mobile/pull/2116/commits/79539637755f2133239ab47f43ff49fcb3b7e6ea

**Comment by aliseyalvi** [comment]
@noumantahir please check the latest commit. Added fix according to your review https://github.com/ecency/ecency-mobile/pull/2116/commits/79539637755f2133239ab47f43ff49fcb3b7e6ea

**Comment by noumantahir** [comment]
seems alright now...

**Review by noumantahir — APPROVED** [review]
seems alright to me now.

**Comment by noumantahir** [comment]
@feruzm awaiting your confirmation...

**Pull request merged**